packages/tex-to-myst/src/math.ts

```diff
@@ -37,4 +39,33 @@ export const MATH_HANDLERS: Record<string, Handler> = {
   'env_vmatrix*': mathBlockAsOriginal,
   env_Vmatrix: mathBlockAsOriginal,
   'env_Vmatrix*': mathBlockAsOriginal,
+  env_subequations(node, state) {
+    state.closeParagraph();
+    state.openNode('mathGroup');
+    state.renderChildren(node);
+    const mathGroup = state.top();
+    transformSubEquations(mathGroup);
+    state.closeNode();
+  },
 };
+
+function transformSubEquations(node: GenericNode) {
+  const mathNodes = selectAll('math', node) as Math[];
+  node.equations = mathNodes.map((math) => math.value);
+  const children = mathNodes
+    .map((math) => {
+      const tex = math.value.trim();
+      if (!tex.match(/^\\begin\{align\*?\}/)) return math;
+      return tex
+        .replace(/^\\begin\{align\*?\}/, '')
```

Comment:
This seems good for now. Will there be problems if only some of the subequations are labeled in latex?

Reply:
I guess this would be something like extending the `labelMathNodes` functionality down to this transform... rather than just enumerating them all later.

Reply:
The label would be consumed by the sub-equation scope, so if would still start with the `\begin{align}` here I think!

The labelMathNodes transform then does pull these labels out.